proxy/http2/Http2Stream.h

```diff
@@ -249,6 +251,8 @@ class Http2Stream : public ProxyClientTransaction
 
   ChunkedHandler chunked_handler;
   Event *cross_thread_event = nullptr;
+  int reentrancy_count      = 0;
+  bool terminate_stream     = false;
```

Comment:
It might be possible to multi-value the `closed` flag, but fundamentally you have 3 states that must be tracked.

* open
* should be closed / prospective close.
* actually closed.

The middle state is the point of this PR, to mark the intent to close when the stream cannot yet be safely closed (waiting for stack unwind).